plugins/modules/dcnm_vrf.py

```diff
@@ -369,24 +456,73 @@ def update_attach_params(self, attach, vrf_name, deploy, vlanId=''):
 
+        ext_values = {}
+        if attach['vrf_lite']:
+            '''Before apply the vrf_lite config, need double check if the swtich role is started wth Border'''
+            if (role.lower() != 'border' and
+                role.lower() != 'border spine' and
+                role.lower() != 'border gateway' and
+                role.lower() != 'border gateway spine' and
+                role.lower() != 'border super spine' and
+                role.lower() != 'border gateway super spine'):
+                msg = 'VRF LITE cannot be attached to switch {} with role {}'.format(attach['ip_address'], role)
+                self.module.fail_json(msg=msg)
+
+            at_lite = attach['vrf_lite']
+            for a_l in at_lite:
+                if a_l['interface'] and a_l['dot1q'] and a_l['ipv4_addr'] and a_l['neighbor_ipv4'] and a_l['ipv6_addr'] \
+                    and a_l['neighbor_ipv6'] and a_l['peer_vrf']:
+
+                    ''' if all the elements are provided by the user in the playbook fill the extension values'''
+                    vrflite_con={}
+                    vrflite_con['VRF_LITE_CONN']=[]
+                    vrflite_con['VRF_LITE_CONN'].append({})
+
+                    vrflite_con['VRF_LITE_CONN'][0]['IF_NAME']=a_l['interface']
+                    vrflite_con['VRF_LITE_CONN'][0]['DOT1Q_ID']=str(a_l['dot1q'])
+                    vrflite_con['VRF_LITE_CONN'][0]['IP_MASK']=a_l['ipv4_addr']
+                    vrflite_con['VRF_LITE_CONN'][0]['NEIGHBOR_IP']=a_l['neighbor_ipv4']
+                    vrflite_con['VRF_LITE_CONN'][0]['NEIGHBOR_ASN']='65535'
+                    vrflite_con['VRF_LITE_CONN'][0]['IPV6_MASK']=a_l['ipv6_addr']
+                    vrflite_con['VRF_LITE_CONN'][0]['IPV6_NEIGHBOR']=a_l['neighbor_ipv6']
+                    vrflite_con['VRF_LITE_CONN'][0]['AUTO_VRF_LITE_FLAG']='false'
+                    vrflite_con['VRF_LITE_CONN'][0]['PEER_VRF_NAME']=a_l['peer_vrf']
+                    vrflite_con['VRF_LITE_CONN'][0]['VRF_LITE_JYTHON_TEMPLATE']='Ext_VRF_Lite_Jython'
+                    ext_values['VRF_LITE_CONN']=json.dumps(vrflite_con)
+
+                    ms_con={}
+                    ms_con['MULTISITE_CONN']=[]
+                    ext_values['MULTISITE_CONN']=json.dumps(ms_con)
+
+                    self.vrflitevalues = ext_values
+                    self.vrf_ext = True
+
         attach.update({'fabric': self.fabric})
         attach.update({'vrfName': vrf_name})
         attach.update({'vlan': vlanId})
         attach.update({'deployment': deploy})
         attach.update({'isAttached': deploy})
         attach.update({'serialNumber': serial})
-        attach.update({'extensionValues': ""})
-        attach.update({'instanceValues': ""})
+        if self.vrf_ext:
+            attach.update({'extensionValues': json.dumps(ext_values).replace(' ','')})
+            attach.update({'instanceValues': "{\"loopbackId\":\"\",\"loopbackIpAddress\":\"\",\"loopbackIpV6Address\":\"\"}"})
```

Comment:
Can you use single quotes here so you don't have to escape `\` the double quotes `\"` ?

Reply:
We are just adding this line straight from how the DCNM displays the instance values, we are not adding or verifying the same anywhere, hence planning to leave it as it is.